### ISSUE TITLE
[Bug] Property 'getCoordinatePrecision' is missing in type 'TerraDrawMapLibreGLAdapter'

### ISSUE THREAD

**Issue by ingalls** [JamesLMilner/terra-draw#248]
**Describe the bug**

@JamesLMilner When using TerraDraw in a Vite/Vue project with typescript checking provided by `vue-tsc` the following error is thrown during compilation

```
src/stores/map.ts:534:17 - error TS2741: Property 'getCoordinatePrecision' is missing in type 'TerraDrawMapLibreGLAdapter' but required in type 'TerraDrawAdapter'.

534                 adapter: new terraDraw.TerraDrawMapLibreGLAdapter({ map: this.map }),
                    ~~~~~~~

  node_modules/terra-draw/dist/common.d.ts:94:5
    94     getCoordinatePrecision(): number;
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    'getCoordinatePrecision' is declared here.
  node_modules/terra-draw/dist/terra-draw.d.ts:44:9
    44         adapter: TerraDrawAdapter;
               ~~~~~~~
    The expected type comes from property 'adapter' which is declared here on type '{ adapter: TerraDrawAdapter; modes: TerraDrawBaseDrawMode<any>[]; idStrategy?: IdStrategy<FeatureId> | undefined; tracked?: boolean | undefined; }'
```

When calling the following code:

```
adapter: new terraDraw.TerraDrawMapLibreGLAdapter({ map: this.map })
```

**Terra Draw npm version**

> 0.0.1-alpha.65

**To Reproduce**

I can work on a reproducible test case if needed.

**Expected behavior**

That the MapLibreGLAdapater would provide all functions to confirm to the TerraDrawAdapter interface

**Screenshots**

N/A

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Browser: N/A - occurs during TS compile
 - Version 24.04
 
**Smartphone (please complete the following information):**
N/A - occurs during compilation check

**Additional context**

Vue-TSC / Typescript dependencies

```
├─┬ pinia@2.1.7
│ └── typescript@5.4.5
├─┬ vue-tsc@2.0.16
│ ├─┬ @vue/language-core@2.0.16
│ │ └── typescript@5.4.5 deduped
│ └── typescript@5.4.5 deduped
└─┬ vue@3.4.27
  └── typescript@5.4.5 deduped
```

**Comment by JamesLMilner** [comment]
Hey @ingalls - thanks for this. Let me try and help you here!

I am a bit confused as `getCoordinatePrecision` comes from the base class that all adapters extend from, and should be present in all adapters. Let me do a bit of digging and report back.

**Comment by JamesLMilner** [comment]
I can't reproduce this locally in the `development` folder. The build there is powered by Webpack. I am wondering if it's something specific about `vue-tsc` ? A reproduction case might be helpful.

**Comment by ingalls** [comment]
@JamesLMilner Gladly, I really appreciate you spending the time to look into this for me.

I've build the smallest reproducible test case I possibly could here:
https://stackblitz.com/edit/vitejs-vite-6likzu?file=src%2FApp.vue

- The only file is really relevant is the `src/App.vue`
- Typescript compilation by `vue-tsc` can be initiation with `npm run build` and will output the same error as above.

I've also attached a static version in a zip to this issue
[vitejs-vite-6likzu.zip](https://github.com/JamesLMilner/terra-draw/files/15249786/vitejs-vite-6likzu.zip)

**Comment by JamesLMilner** [comment]
Thanks for sharing the reproduction - I'll try and get to this as soon as I get a moment.

**Comment by JamesLMilner** [comment]
@ingalls thanks for the reproduction case - super helpful. I was able to see the error. However, I am a bit unsure as to why it is happening, as the abstract class that it extends from `TerraDrawBaseAdapter` already has the concrete implementation for `unregister` and `getCoordinatePrecision`. I  have a gut feeling that this might go away if we added:

```
	public getCoordinatePrecision(): number {
		return super._coordinatePrecision;
	}
```

On the parent adapters, but this shouldn't be necessary to my understanding. My gut instinct is this might be a `vue-tsc` issue but I can't be 100% sure - I might see if I can get just a plain tsc build working locally.

**Comment by JamesLMilner** [comment]
Okay, I can replicate with plain `tsc` so this is not isolated to `vue-tsc`. Give me some time to resolve this as I can imagine this will be a pain point for a lot of people

**Comment by JamesLMilner** [comment]
@ingalls this is now resolved in `0.0.1-alpha.66` 🎉 I don't fully understand why this didn't work - it may be something to do with microbundle, but it was too time consuming to keep debugging. If anyone reading this knows I'd be happy to learn more!

Anyhow, very much appreciate you raising this and glad we could get it fixed. Would be interested to hear what you're working on - let me know if there's any other issues you hit along the way!

**Comment by JamesLMilner** [comment]
Hey @ingalls no problem, I'm glad we could get it resolved! The reproduction case and all the info you provided made it really smooth to figure out what was happening.

That's really cool to hear - thank you for taking the time to share that with me. Always nice to hear when your OSS libraries are powering real applications! 

If you find any more issues, or have ideas for new features, please feel free to raise them on the issue tracker.
